walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,91 @@
 // limitations under the License.
 
 package com.google.sps;
-
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
 
+/**
+ * Returns possible meeting times on a certain day given a meeting request.
+ */
 public final class FindMeetingQuery {
+
+  public static final int MINUTES_IN_DAY = 1440;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> ret = new ArrayList<TimeRange>();
+    long duration = request.getDuration();
+
+    // no options for too long of a request
+    if (duration > MINUTES_IN_DAY){
+      return ret;
+    }
+
+    Set<String> mandatoryAttendees =  new HashSet<String>();
+    mandatoryAttendees.addAll(request.getAttendees()); 
+
+    Set<String> optionalAttendees = new HashSet<String>();
+    optionalAttendees.addAll(request.getOptionalAttendees());
+
+    Set<String> allAttendees = new HashSet<String>();
+    allAttendees.addAll(mandatoryAttendees);
+    allAttendees.addAll(optionalAttendees);
+
+    // options for no attendees
+    if (allAttendees.isEmpty()) {
+      ret.add(TimeRange.WHOLE_DAY);
+      return ret;
+    }
+
+    // find free times
+    List<TimeRange> freeTimes = new ArrayList<TimeRange>(findFreeTimes(allAttendees, events, duration));
+  
+
+    if (freeTimes.isEmpty()) {
+      // adhere to additional test requirement
```

Comment:
someone new reading this code might not know what the additional test requirement is. better to explicitly say what the requirement is